### ISSUE TITLE
MagiskManager 5.1.0 "not responding"

### ISSUE THREAD

**Issue by JERW86** [topjohnwu/MagiskManager#225]
@topjohnwu 
I updated to 13.3 and found that Magisk Manager won't open. When I open it, it won't move past the initial logo and after a minute or so says, "Magisk Manager is not.reponding". I have tried clearing app data and uninstall/reinstalling the app, but it still freezes at the logo screen and gives me the same message. Also see screenshot. For now, I have rolled to 5.0.6. Thank you for all your hard work.

Phone: XT1625 (Moto G 2016 U.S. Non-Amazon)/ROM: GZR Tesla-athene-7.1.2. OFFICIAL-20170611/Android Version: 7.1.2/Security Patch Level: June 05, 2017/Kernel: ElementalX-G4-1.08-LOS/Magisk v13.3 with Magisk Manager 5.0.6/MagiskSU/microG GmsCore v0.2.4-103/NanoMod Patcher 9.2-20170610/Google Play Store/Recovery: Unofficial TWRP athene 3.1.1_shreps

[Magisk Manager 5.1.0 Not Responding.txt](https://github.com/topjohnwu/MagiskManager/files/1157027/Magisk.Manager.5.1.0.Not.Responding.txt) 
Edit: added more logs
Edit 2: forgot to mention I flashed my clean bout.img in recovery before flashing v13.3.
[last_magisk.log.txt](https://github.com/topjohnwu/MagiskManager/files/1157230/last_magisk.log.txt)
[magisk.log.txt](https://github.com/topjohnwu/MagiskManager/files/1157231/magisk.log.txt)
[Full Log.txt](https://github.com/topjohnwu/MagiskManager/files/1157239/Full.Log.txt)

![screenshot_20170718-144008](https://user-images.githubusercontent.com/16406776/28335026-0260de08-6bcb-11e7-9880-15ff8cf5c83f.png)

**Comment by visualfanatic** [comment]
```
E/ActivityManager: ANR in com.topjohnwu.magisk
Reason: executing service com.topjohnwu.magisk/.services.UpdateCheckService
```

**Comment by coderobe** [comment]
Same issue on my Lenovo P2 running Android 7.1.2. Here's a stacktrace:

```
   937.249  1645  4590 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.topjohnwu.magisk/.SplashActivity bnds=[230,228][437,512] (has extras)} from uid 10041 on display 0
   937.301  1645  5476 I ActivityManager: Start proc 13739:com.topjohnwu.magisk/u0a181 for activity com.topjohnwu.magisk/.MainActivity
   938.276 13739 13739 E AndroidRuntime: FATAL EXCEPTION: main
   938.276 13739 13739 E AndroidRuntime: Process: com.topjohnwu.magisk, PID: 13739
   938.276 13739 13739 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.topjohnwu.magisk/com.topjohnwu.magisk.MainActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.Object[] java.util.Collection.toArray()' on a null object reference
   938.276 13739 13739 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2699)
   938.276 13739 13739 E AndroidRuntime:        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2766)
   938.276 13739 13739 E AndroidRuntime:        at android.app.ActivityThread.-wrap12(ActivityThread.java)
   938.276 13739 13739 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1507)
   938.276 13739 13739 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:102)
   938.276 13739 13739 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:154)
   938.276 13739 13739 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6236)
   938.276 13739 13739 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
   938.276 13739 13739 E AndroidRuntime:        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:891)
   938.276 13739 13739 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:781)
   938.276 13739 13739 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.Object[] java.util.Collection.toArray()' on a null object reference
   938.276 13739 13739 E AndroidRuntime:        at java.util.ArrayList.addAll(ArrayList.java:562)
   938.276 13739 13739 E AndroidRuntime:        at com.topjohnwu.magisk.MagiskFragment.ac(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at com.topjohnwu.magisk.MagiskFragment.a(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at android.support.v4.app.Fragment.b(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at android.support.v4.app.m.a(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at android.support.v4.app.m.e(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at android.support.v4.app.m.a(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at android.support.v4.app.m.e(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at android.support.v4.app.m.n(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at android.support.v4.app.j.f(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at android.support.v4.app.h.onStart(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at android.support.v7.app.e.onStart(Unknown Source)
   938.276 13739 13739 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1249)
   938.276 13739 13739 E AndroidRuntime:        at android.app.Activity.performStart(Activity.java:6701)
   938.276 13739 13739 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2662)
   938.276 13739 13739 E AndroidRuntime:        ... 9 more
   938.282  1645  4995 W ActivityManager:   Force finishing activity com.topjohnwu.magisk/.MainActivity
   938.792  1645  1700 W ActivityManager: Activity pause timeout for ActivityRecord{9d5075b u0 com.topjohnwu.magisk/.MainActivity t2666 f}
   938.799  1645  1700 I WindowManager: Failed to capture screenshot of Token{3540bf8 ActivityRecord{9d5075b u0 com.topjohnwu.magisk/.MainActivity t2666 f}} appWin=Window{48d4e3 u0 Starting com.topjohnwu.magisk} drawState=4
   949.054  1645  1700 W ActivityManager: Activity destroy timeout for ActivityRecord{9d5075b u0 com.topjohnwu.magisk/.MainActivity t2666 f}
  1016.848  1645  4995 I ActivityManager: Killing 13739:com.topjohnwu.magisk/u0a181 (adj 900): crash
```

**Comment by coderobe** [comment]
And here's another piece of log after force-stopping & restarting the app:

```
  1283.105  1645  4594 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.topjohnwu.magisk/.SplashActivity bnds=[230,228][437,512] (has extras)} from uid 10041 on display 0
  1283.136  1645  1656 I ActivityManager: Start proc 14201:com.topjohnwu.magisk/u0a181 for activity com.topjohnwu.magisk/.SplashActivity
  1284.690  1645  5078 I ActivityManager: START u0 {cmp=com.topjohnwu.magisk/.MainActivity} from uid 10181 on display 0
  1297.697  1645  1645 E JobServiceContext: Time-out while trying to bind 8826fad #u0a181/1 com.topjohnwu.magisk/.services.UpdateCheckService, dropping.
  1317.700  1645  1700 W ActivityManager: Timeout executing service: ServiceRecord{4777c3a u0 com.topjohnwu.magisk/.services.UpdateCheckService}
  1321.314  1645  1700 E ActivityManager: ANR in com.topjohnwu.magisk
  1321.314  1645  1700 E ActivityManager: PID: 14201
  1321.314  1645  1700 E ActivityManager: Reason: executing service com.topjohnwu.magisk/.services.UpdateCheckService
```

**Comment by Ante0** [comment]
Same issue here.

**Comment by shinji257** [comment]
I can reproduce this as well.  I am at the opinion that this might not be a fault in MagiskManager 5.1.0 though because I reverted to 5.0.6 and the issue persists.  This appears to be linked to the Magisk 13.3 update because it didn't start until after I installed that and rebooted.  Until then 5.1.0 actually loaded fine.

For the sake of troubleshooting I did make sure the following was true/completed
* I cleared cache
* I also tried to uninstall/reinstall
* I made sure it was located on internal storage (I have adoptable storage enabled)

**Comment by JERW86** [comment]
@shinji257 hmm interesting.. for me, reverting back to 5.0.6 with 13.3 installed works fine.

**Comment by shinji257** [comment]
I did the last ditch (to me) troubleshooting step.  Mostly because uninstalling Magisk then rebooting has a risk of losing TWRP until I go through some hoops to put it back.

In TWRP I ran the uninstaller (restore boot image) then reinstalled 13.3.  The only reboot was after the reinstall.  After that it is now working including Magisk Manager 5.1.0.  Keeping in mind that all modules that I had setup were also removed  in the process.  It is possible that one of the modules was causing the issue.

Modules installed are as follows:
Dolby Atmos r6.5-c
Universal deep_buffer Remover v2.4
Audio Modification Library v1.4
A couple of self-made modules for toggling on the multi-user mode and adoptable storage interface on my LG G5.

**Comment by Ante0** [comment]
Tried the following:
Uninstalled Magisk by flashing back stock boot image, then flashing magisk again. Install manager 5.1.0, freezing at start. Uninstalled manager and installed 5.0.6 again, working. 
Uninstalled manager 5.0.6 then installed 5.1.0, freezing. Uninstalled 5.1.0, installed 5.0.6. Disabled all modules then installed 5.1.0 over it, freezing. So it doesn't seem to be module related.

**Comment by Ante0** [comment]
@shinji257 using Uninstaller (and rebooting recovery as every flash afterward ended with 255)then reflashing 13.3 seems to fix it. Down side is that /data/magisk.img is removed so you have to set up modules again. But works with 5.1.0 now.

**Comment by Shadowxp13** [comment]
I'm with an similar issue. If i install any magisk manager apk higher than 5.0.1 my phone goes mad and everything starts crashing. The curious thing is that i also have the dolby atmos mod, and as it is an requirement AudModLib. I'll try uninstalling everything now

**Comment by coderobe** [comment]
@Shadowxp13 that is probably unrelated to this specific issue - try to keep the threads on topic

**Comment by roshanravan** [comment]
Same not responding with Latest Magisk Manager Version: v5.1.0
I could open it when I disabled wifi and data reboot deviceand magisk manager opens. I Close it and as soon as I turn wifi or data on it won't open. Even turning off wifi and data won't help after that

**Comment by unrealbe** [comment]
Same issue, reverting to 5.0.6 solved it. 

Modules: Greenify, hosts adblock and youtube

**Comment by roshanravan** [comment]
Yes reverting to 5.0.6 solved it

**Comment by FabianMeul** [comment]
Also crashing on Nexus 6P. (5.1.0)

Just wondering; how does one revert to 5.0.6?
When I download the APK from the "releases" section, it says the APK is corrupted...

**Comment by Ante0** [comment]
@FabianMeul you need to uninstall 5.1.0 first.

**Comment by FabianMeul** [comment]
Confirmed working when reverted to 5.0.6.
(Thx @Ante0 for the revert instructions)

**Comment by roshanravan** [comment]
i suspect the issue might be in Marshmallow Cause my friend with same device as mine has Nougat and Magisk Manager 5.1.0 works Ok.

**Comment by FabianMeul** [comment]
@roshanravan I have Android 7.1.2 (Nougat) and Magisk Manager 5.1.0 crashes.

**Comment by Ante0** [comment]
@roshanravan I'm on nougat and have problems.
I think it's /data/magisk.img as @shinji257 said, possibly some module.

**Comment by roshanravan** [comment]
@Ante0 oh so its not Nougat but my modules are different with people who have this problem

**Comment by Ante0** [comment]
Then it's probably how the magisk.img is built. Manager 5.1.0 might read it differently.

**Comment by pizzadude** [comment]
See #231 for a temporary fix.

_Sent from my Motorola Moto G 2014 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by SurenSeven** [comment]
Can confirm too that https://github.com/topjohnwu/MagiskManager/issues/231 fixed the issue. Running PN on 6P.

**Comment by JERW86** [comment]
Fixed in Magisk Manager 5.1.1.
